### PR TITLE
Collection name to CamelCase (fixes #2983)

### LINKED ISSUE #2983: Create generic method to change collection name to camel case on client side
See discussion in #2973. The `formName.charAt(0).toUpperCase() + formName.slice(1) + 'Form'` lines and the so called hotfixes have to be refactored.

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Also fixes topic detail state url from `topics/topic/2` to `topics/2`; See all other apps.

### CODE REVIEW

File: openslides/core/static/js/core/base.js [hunk 1]
```diff
@@ -953,6 +948,33 @@ angular.module('OpenSlidesApp.core', [
     }
 ])
 
+/* Converts a snake-case string to CamelCase. */
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: The comment is not completely correct: `foo-bar` is converted to `fooBar` and not to `FooBar`. May be add a correct example to show what is going on here.

File: openslides/core/static/js/core/base.js [hunk 2]
```diff
@@ -953,6 +948,33 @@ angular.module('OpenSlidesApp.core', [
...
+.factory('CamelCase', [
+    function () {
+        return function (str) {
+            return str.replace(/-([a-z])/g, function (g) {
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: The variable `g` is not self-explaining. Maybe use something like `fragment` or `part` or `match`

File: openslides/core/static/js/core/base.js [hunk 3]
```diff
@@ -953,6 +948,33 @@ angular.module('OpenSlidesApp.core', [
...
+
+/* Return the specific EditForm for a given model. */
+.factory('EditForm', [
+    '$injector',
+    'CamelCase',
+    function ($injector, CamelCase) {
+        return {
+            fromCollectionString: function (collection) {
+                var modelName = CamelCase(collection).split('/')[1];
+                // Contert modelModel to ModelModelForm
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Typo.

File: openslides/core/static/js/core/base.js (hunk 1 above)
Comment (FinnStutzenstein) [inline openslides/core/static/js/core/base.js]: Done.

File: openslides/core/static/js/core/base.js (hunk 2 above)
Comment (FinnStutzenstein) [inline openslides/core/static/js/core/base.js]: Done.

File: openslides/core/static/js/core/base.js (hunk 3 above)
Comment (FinnStutzenstein) [inline openslides/core/static/js/core/base.js]: Done.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Ping.

**Comment by emanuelschuetze** [comment]
Rebased.

**Pull request merged**